### ISSUE TITLE
linter: curly set to multi-line triggers false positive on single line else

### ISSUE THREAD

**Issue by MichaelGoff** [oxc-project/oxc#9600]
### What version of Oxlint are you using?

0.15.13

### What command did you run?

`npx oxlint`

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "env": {
    "browser": true
  },
  "categories": {
    "correctness": "error"
  },
  "rules": {
    "curly": ["error", "multi-line"]
  }
}
```

### What happened?

The following snippet passes eslint when set to `curly: ['error', 'multi-line']` but fails in oxlint. 

I expect this single line else to be okay when set to `'multi-line'`

```ts
  const isIterable = (obj: any) : obj is Iterable<IgnoreRule> => {
    if (obj === null) return false;
    else if (typeof obj === 'string') return false;
    else return typeof value[Symbol.iterator] === 'function';
  };
```

Oxlint output:

```
  × eslint(curly): Expected { after 'else'.
    ╭─[src/testFile.ts:88:10]
 87 │     else if (typeof obj === 'string') return false;
 88 │     else return typeof value[Symbol.iterator] === 'function';
    ·          ────────────────────────────────────────────────────
 89 │   };
    ╰────
  help: Replace `return typeof value[Symbol.iterator] === 'function';` with `{return typeof value[Symbol.iterator] === 'function';}`.
```

**Comment by heygsc** [comment]
It's a bit strange, I tried testing with or without parentheses in the source code, but both cargo tests passed.

**Comment by MichaelGoff** [comment]
Here's a minimal reproduction https://github.com/MichaelGoff/oxlint-curly-multiline-issue 

`npx oxlint` throws the error

I'm on Windows 10 using Node 22.11.0 if that's relevant.

**Comment by heygsc** [comment]
mac arm64 + node 22.14.0, `Found 0 warnings and 0 errors.`.

Maybe someone else will try again?

**Comment by camc314** [comment]
fixed by #10671
